### ISSUE TITLE
basename for use on GitHub Pages

### ISSUE THREAD

**Issue by MrLoh** [FormidableLabs/redux-little-router#129]
Hi, I love the idea behind Redux Little Router, I have a little problem getting it to work with GitHub Pages. I am using Create React App and its deploy script to publish to GitHub pages. My app lives at  `https://user.github.io/Repo` so I set `basename: Repo` in the hopes that would make the app work, but somehow that didn't solve the issue. On my lokal development server, the app still starts at `http://localhost:3000` instead of `http://localhost:3000/Repo`, Once I click a link it works correctly, navigating to `http://localhost:3000/Repo/route` and the `pathname` in the state is `page`. But when I reload the page on `http://localhost:3000/Repo/route` the `pathname` in the state becomes `Repo/page` instead of just `page` and hence nothing is matched. 

Am I misunderstanding how basename is supposed to work, or is this a bug?

**Comment by MrLoh** [comment]
I realized that GitHub Pages has some general problems with `browserHistory`, but even (this workaround)[https://github.com/rafrex/spa-github-pages] had no effect. I have the feeling that there is a bug in the implementation in how URLs are parsed, that it doesn't take care of the `basename`.

**Comment by karenling** [comment]
Having a similar issue. I was able to reproduce it on the demo app.

I just added `basename: '/something'` in [`demo/client/app.js`](https://github.com/FormidableLabs/redux-little-router/blob/master/demo/client/app.js). i.e.:
```
const {
  routerEnhancer,
  routerMiddleware
} = routerForBrowser({ routes, basename: '/something' });
```

![basename](https://cloud.githubusercontent.com/assets/4053800/21463519/48e8cb0a-c91f-11e6-8c66-ab8aa67d15f3.gif)

Notice how clicking any of the links works fine, but if I refresh the page (or go directly to `http://localhost:4567/something/cat`), none of the routes match anymore.

Any clues on what I may be doing wrong or is this a bug?

**Comment by NeoPhi** [comment]
This is a bug in how the state is initialized. If you look at https://github.com/FormidableLabs/redux-little-router/blob/master/src/link.js you'll notice that it manipulates `basename` for the location that ultimately gets pushed. However https://github.com/FormidableLabs/redux-little-router/blob/master/src/store-enhancer.js initialization of the location doesn't take basename into account.

**Comment by tptee** [comment]
Should be fixed in `v13.0.0`! Ping here if not 😬
